### DISCUSSION TITLE
Is there a way to retrieve the list of asset groups in Python?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17682] [Q&A]
Hey,
is there any way to  retrieve the `asset group list`? I plan to create jobs iteratively (one job per group)

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699004303216979?thread_ts=1699004303.216979&cid=C01U954MEER).*

**Comment by sryza** [comment]
You can collect the set of all asset groups by iterating over the asset definitions:

```python
from dagster import asset


@asset(group_name="group1")
def asset1():
    ...


@asset(group_name="group1")
def asset2():
    ...


@asset(group_name="group2")
def asset3():
    ...


assets = [asset1, asset2, asset3]

all_group_names = {
    group_name for asset_def in assets for group_name in asset_def.group_names_by_key.values()
}
```
